File: python/evals/autocontext_fixture_js/src/auth/middleware.ts
// Auth middleware.
import { SessionManager } from "./session";
import { logger } from "../utils/logging";

const mgr = new SessionManager();

export function authMiddleware(req: any) {
  const token = (req.headers?.authorization || "").replace("Bearer ", "");
  const session = mgr.get(token);
  if (!session) {
    logger.warn("unauthenticated request");
    return [{ error: "unauthorized" }, 401];
  }
  req.session = session;
  return null;
}

File: python/evals/autocontext_fixture/src/auth/middleware.py
"""auth_middleware — validates session token on every request."""
from src.auth.session import SessionManager
from src.utils.logging import setup_logger

logger = setup_logger(__name__)


def auth_middleware(request):
    token = request.headers.get("Authorization", "").replace("Bearer ", "")
    manager = SessionManager()
    session = manager.get(token)
    if not session:
        logger.warning("unauthenticated request")
        return {"error": "unauthorized"}, 401
    request.session = session
    return None
